### PR TITLE
Features/12/git hooks

### PR DESCRIPTION
**Pull request by what-the-hub**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [x] @github_username

## Summary of issue

ToDo

Added Husky and Lint-staged

ToDo

in progress

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -36,10 +37,23 @@
...
     "sass-loader": "^8.0.2",
     "typescript": "~4.1.5",
     "vue-jest": "^5.0.0-0"
+  },
+  "lint-staged": {
+    "*.{html,vue,css,scss}": [
+      "prettier --write",
+      "git add"
+    ],
+    "*.{js,vue}": [
```
Comment (maveric-ui) [inline package.json]: plz, add for .ts files as well

File: package.json [hunk 2]
```diff
@@ -36,10 +37,23 @@
...
+      "eslint --fix"
+    ],
+    "*": "prettier --ignore-unknown --write",
+    "*.js": "eslint --cache --fix"
```
Comment (maveric-ui) [inline package.json]: plz, add for .ts files as well

**Comment by maveric-ui** [comment]
plz, add lint-staged on pre-commit hook
"husky": { 
   "hooks": {
      "pre-commit": "lint-staged && yarn run test:unit"
   }
 }

File: package.json (hunk 1 above)
Comment (what-the-hub) [reply package.json]: ok

File: package.json [hunk 3]
```diff
@@ -38,10 +39,28 @@
...
+      "prettier --write",
+      "git add"
+    ],
+    "*.{js,ts,vue}": [
+      "eslint --fix"
+    ],
+    "*": "prettier --ignore-unknown --write",
+    "*.{js,ts}": "eslint --cache --fix"
+  },
+  "husky": {
```
Comment (maveric-ui) [inline package.json:61]: plz, fix align

**Review by maveric-ui — DISMISSED** [review]

File: package.json [hunk 4]
```diff
@@ -38,10 +39,28 @@
...
+    "hooks": {
+    "pre-commit": "lint-staged && yarn run test:unit"
```
Comment (maveric-ui) [inline package.json]: remove '&& yarn run test:unit'
don't need to run test as well we don't use yarn

**Review by maveric-ui — APPROVED** [review]

**Review by markuwas — APPROVED** [review]

**Review by tsehla42 — APPROVED** [review]

**Pull request merged**